### DISCUSSION TITLE
Github Actions cache acting bizarre? Cache hits not being found

### DISCUSSION THREAD

**Discussion by cullenwren-volair** [community/community#58583] [Actions]
### Select Topic Area

Question

### Body

Is anyone else having issues with Github Actions caching? For my workflow which has not changed, I am not able to get cache hits and each branch is uploading a new cache with the same key.

**Comment by piotrekkr** [comment]
Not one hundred percent sure but cache in most cases is not shared between branches. If cache key is not found on current branch it will search current branch cache using `restore-keys`. If there is also no match it will search default repository branch cache using key and then `restore-keys`. Here is some [documentation](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache) 

Based on that each new branch almost always starts with no cache (exceptions in docs) and then tries to find cache key on default repo branch. At least this is my understanding of this.

**Comment by cullenwren-volair** [comment]
Yup, that all makes sense. From the docs:

> Workflow runs can restore caches created in either the current branch or the default branch (usually main).

However what I'm experiencing is the inability for the branches to restore `main`'s cache.

**Reply by piotrekkr** [reply]
Are you sure there is actually a cache key matching on `main` branch? There is a way to display caches using gh cli or by using github UI in Actions > Caches (`https://github.com/<org>/<repo>/actions/caches?query=branch%3Amain`)

**Reply by cullenwren-volair** [reply]
Yes, here's a screen cap of the caches: 
![image](https://github.com/community/community/assets/124718675/dcbecc26-842f-4873-83dd-7db74d8697e2)

The `main` cache has "cached"  and "last used" of 20 minutes because of a merge with main, however it existed before and after was still not being used by the CI. Quite perplexed

**Comment by cullenwren-volair** [comment]
Well, didn't change anything and it's working again now.... 🤷🏻‍♂️

**Reply by MatMercer** [reply]
Same thing happened to me at jul 2023, now I'm waiting for the same thing to happen to me again 😅 .

Everything is right, cache is at default branch, PR workflow just can't match it.

**Comment by kkew3** [comment]
Same thing happened to me around May 25, 2026. I waited a week for it to work with no luck. My solution was to delete the cache and let it regenerate. After that, the action picked up the regenerated cache correctly.
